OverlayPlugin.Core/EventSources/FFXIVRequiredEventSource.cs

```diff
@@ -369,43 +376,133 @@ public override void SaveConfig(IPluginConfig config)
+            {
+                dispatchEvent = true;
+            }
+            // Check each of party and alliances A/B/C/D/E
+            if (!dispatchEvent)
+            {
+                dispatchEvent = HasPartyCompChanged(cachedPartyList.partyMembers, newParty.partyMembers);
+            }
+            if (!dispatchEvent)
+            {
+                dispatchEvent = HasPartyCompChanged(cachedPartyList.allianceAMembers, newParty.allianceAMembers);
+            }
+            if (!dispatchEvent)
+            {
+                dispatchEvent = HasPartyCompChanged(cachedPartyList.allianceBMembers, newParty.allianceBMembers);
+            }
+            if (!dispatchEvent)
+            {
+                dispatchEvent = HasPartyCompChanged(cachedPartyList.allianceCMembers, newParty.allianceCMembers);
+            }
+            if (!dispatchEvent)
+            {
+                dispatchEvent = HasPartyCompChanged(cachedPartyList.allianceDMembers, newParty.allianceDMembers);
+            }
+            if (!dispatchEvent)
+            {
+                dispatchEvent = HasPartyCompChanged(cachedPartyList.allianceEMembers, newParty.allianceEMembers);
+            }
+            cachedPartyList = newParty;
+            if (dispatchEvent)
+            {
+                DispatchPartyChangeEvent();
+            }
+        }
+
+        private bool HasPartyCompChanged(PartyListEntry[] oldList, PartyListEntry[] newList)
+        {
+            // If the old list was null and the new list isn't null, they've changed, dispatch the event
+            if (oldList == null && oldList != newList)
+            {
+                return true;
+            }
+
+            for (var i = 0; i < newList.Length; ++i)
+            {
+                var newMember = newList[i];
+                var oldMember = oldList[i];
+                if (newMember == null || oldMember == null)
+                {
+                    // If one of these is null and the other isn't, they've changed, dispatch the event
+                    if (newMember != oldMember)
                     {
-                        foreach (var c in combatants)
-                        {
-                            if (missingPartyMembers.Contains(c.ID))
-                            {
-                                missingPartyMembers.Remove(c.ID);
-                            }
-                        }
+                        return true;
                     }
-
-                    // Send an update event once all party members have been found.
-                    if (missingPartyMembers.Count == 0)
+                    // Otherwise they're both null, we don't need to check them, continue
+                    else
                     {
-                        DispatchPartyChangeEvent(cachedPartyList, 0);
+                        continue;
                     }
                 }
+                // If the party list is in a different order, dispatch the event
```

Comment:
I thought this party list wasn't ordered? Or, what does order mean here?

Reply:
There are two potential cases where the order varies:

1. We're processing the current player's party, and someone has left or joined and shuffled the `List` values. This acts as an early out of the loop to avoid processing additional data
2. We're processing an alliance of some sort. These are sparsely populated fixed-size arrays, similarly if there's someone missing or a new player added in place of an old one, this acts as an early out to avoid having to process the whole list.